docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -301,29 +301,42 @@ type TrainJobSpec struct {
 	Suspend *bool `json:"suspend,omitempty"`
 
 	// ManagedBy is used to indicate the controller or entity that manages a TrainJob.
+	// The value must be either an empty, 'training-operator.kubeflow.org/trainjob-controller' or
+	// 'kueue.x-k8s.io/multikueue'.
+	// The built-in TrainJob controller reconciles TrainJob which don't have this
+	// field at all or the field value is the reserved string
+	// 'training-operator.kubeflow.org/trainjob-controller', but delegates reconciling TrainJobs
+	// with a 'kueue.x-k8s.io/multikueue' to the Kueue.
+	//
+	// The value must be a valid domain-prefixed path (e.g. acme.io/foo) -
+	// all characters before the first "/" must be a valid subdomain as defined
+	// by RFC 1123. All characters trailing the first "/" must be valid HTTP Path
+	// characters as defined by RFC 3986. The value cannot exceed 63 characters.
+	// The field is immutable.
 	ManagedBy *string `json:"managedBy,omitempty"`
 }
 
 type TrainingRuntimeRef struct {
-	// Name for the training runtime.
+	// Name for the runtime.
+	// This must indicate the runtime deployed in the same namespace as the TrainJob
+	// when TrainingRuntime is used in the kind.
 	Name string `json:"name"`
 
-	// Namespace for the runtime. In that case, user should use TrainingRuntime
-	Namespace *string `json:"namespace,omitempty"`
+	// APIVersion is the apiVersion for the runtime.
+	// Defaults to the v2alpha1.
```

Comment:
```suggestion
	// Defaults to the kubeflow.org/v2alpha1.
```

I'm wondering if we should align with the YAML apiVersion like this.

Reply:
APIVersion would indeed look more consistent with how objects are declared in yamls [example](https://github.com/kubeflow/training-operator/blob/7631f4817e92ecaec73d66e0d6a6e9e916524919/examples/pytorch/mnist/v1/pytorch_job_mnist_gloo.yaml#L1-L2). OTOH consider if you need a version at all - will you support other versions than the current? Maybe just `APIGroup` is enough.

Reply:
I'm assuming that we need to support multiple versions of TrainingRuntimes when we graduate it from Beta to Stable so that we align with the Kubernetes deprecation policy.

But, I don't think that we need to support multiple versions when we graduate the API from alpha to beta.
So, I'm wondering if we should drop this version field in the alpha and beta stages, and then if we should revisit this discussion to mitigate the GA graduation risks.

@andreyvelich @mimowo WDYT?

Reply:
Agree, let's do it in the future versions.
What are the benefits you see to add the APIGroup in the runtime reference if that will be always `kubeflow.org` ?
To give users possibility to build custom APIs for the training runtimes ?

Reply:
@andreyvelich Oh, I actually still have not added the `apiGroup` field in the KEP.
So, based on their reactions, I opened the PR to add the `apiGroup` in the API design.

Reply:
oh, you are right. Let's add it in the followup PR.